src/overlays/actors/ovl_En_Tanron3/z_en_tanron3.c

```diff
@@ -28,39 +36,413 @@ const ActorInit En_Tanron3_InitVars = {
+    Actor_SetScale(&this->actor, 0.02f);
+    EnTanron3_SetupLive(this, globalCtx);
+    this->actor.flags &= ~1;
+    this->currentRotationAngle = Rand_ZeroFloat(500000.0f);
+    this->waterSurfaceYPos = 430.0f;
+    sGyorg = (Boss03*)this->actor.parent;
+}
+
+void EnTanron3_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    sGyorg->unk_252--;
+}
+
+void EnTanron3_SpawnBubbles(EnTanron3* this, GlobalContext* globalCtx) {
+    static Color_RGBA8 sPrimColor = { 100, 55, 55, 255 };
+    static Color_RGBA8 sEnvColor = { 50, 10, 10, 255 };
+    s32 i;
+    Vec3f velocity;
+    Vec3f acceleration;
+
+    for (i = 0; i < 20; i++) {
+        Matrix_InsertYRotation_f(Rand_ZeroFloat(2 * M_PI), MTXMODE_NEW);
+        Matrix_RotateStateAroundXAxis(Rand_ZeroFloat(2 * M_PI));
+        Matrix_GetStateTranslationAndScaledZ(Rand_ZeroFloat(3.0f) + 2.0f, &velocity);
+        acceleration.x = velocity.x * -0.05f;
+        acceleration.y = velocity.y * -0.05f;
+        acceleration.z = velocity.z * -0.05f;
+        EffectSsDtBubble_SpawnCustomColor(globalCtx, &this->actor.world.pos, &velocity, &acceleration, &sPrimColor,
+                                          &sEnvColor, Rand_ZeroFloat(30.0f) + 70.0f, Rand_ZeroFloat(5.0f) + 15.0f, 0);
+    }
+}
+
+void EnTanron3_SetupLive(EnTanron3* this, GlobalContext* globalCtx) {
+    this->actionFunc = EnTanron3_Live;
+    Animation_MorphToLoop(&this->skelAnime, &D_0600DAAC, -10.0f);
+    this->rotationStep = 0;
+    this->rotationScale = 5;
+    this->workTimer[TANRON3_WORK_TIMER_PICK_DIRECTION] = 50;
+    this->actor.speedXZ = 5.0f;
+    this->speedMaxStep = 0.5f;
+    this->deviationFromCurrentPos.x = randPlusMinusPoint5Scaled(500.0f);
+    this->deviationFromCurrentPos.y = randPlusMinusPoint5Scaled(100.0f);
+    this->deviationFromCurrentPos.z = randPlusMinusPoint5Scaled(500.0f);
+    Math_Vec3f_Copy(&this->currentPos, &this->actor.world.pos);
+    this->timer = Rand_ZeroFloat(100.0f);
+}
+
+/**
+ * This controls the vast majority of the fish's behavior while it's alive, including:
+ * - deciding whether to be hostile or not
+ * - determing whether the fish is beached or not
+ * - swimming towards the player to attack them
+ * - swimming around idly if the player is out of range
+ * - flopping around on land if it beaches itself
+ */
+void EnTanron3_Live(EnTanron3* this, GlobalContext* globalCtx) {
+    s32 atanTemp;
+    f32 xDistance;
+    f32 yDistance;
+    f32 zDistance;
+    f32 xzDistance;
+    f32 extraScaleY = 0.0f;
+    Player* player = GET_PLAYER(globalCtx);
+
+    this->skelAnime.curFrame = 4.0f;
+    if ((player->actor.bgCheckFlags & 1) && player->actor.shape.feetPos[0].y >= 438.0f) {
+        // Player is standing on the central platform, so stop chasing them
+        this->isNonHostile = true;
+    } else if (this->isNonHostile && this->workTimer[TANRON3_WORK_TIMER_WAIT] == 0 && !(this->timer & ((1 << 5) - 1))) {
```

Comment:
Just want to clarify, did you actually mean for the timer ands to be converted to this format @EllipticEllipsis? https://github.com/zeldaret/mm/pull/376#discussion_r763230139

Reply:
No I did not! That was just a way to explain in general.

Reply:
Whoops, that was a misinterpretation on my part then. I'll undo that.